### DISCUSSION TITLE
shortcut to copy URI from UI

### DISCUSSION THREAD

**Discussion by fuglphoenix** [music-assistant/support#2311] [Feature requests and ideas]
Currently the easiest way (as far as I know) to get the URI for a playlist, album etc. is using the search service call.
it would be nice to have an option to extract the URI directly within the MASS UI. 
eg. i would like to automate the execution of a playlist with Coldplay music,   
1. i would use the UI to search or browse my own library of playlists
2. find the playlist i want to automate
3. in the 3 dots menu, have an option to copy the URI string which can be used in the service call mass.play_media

**Comment by OzGav** [comment]
You can find the playlist URI at the bottom of the view in the provider details section.

**Reply by DonGar** [reply]
Excellent!

Though, for whatever reason it's difficult to select or copy that text. I found myself following the link the copying it from the browser afterwards.

**Comment by fuglphoenix** [comment]
It's not the correct URI that is shown. eg.  from the service i get: 
```
  - item_id: VLPL607254AB9BF26EF3
    provider: ytmusic
    name: Queen top100
    version: ""
    sort_name: queen top100
    uri: ytmusic://playlist/VLPL607254AB9BF26EF3
    external_ids: []
    media_type: playlist
    position: null
    owner: Joe DOW
    is_editable: false
```
but in the UI i get: PL607254AB9BF26EF3.  
I see that in most cases it is just missing the beginning VL... i could manage to correct that manually, but another issue is that i selecting the uri-text is disabled, which is really annoying when it is 20+ chars long.

it also not showing the correct uri, for the playlist "my likes"

**Reply by OzGav** [reply]
Ok we will have a look at this further

**Reply by marcelveldt** [reply]
can you add this one as task on the backlog ? Nice small feature for 2.1

**Reply by DonGar** [reply]
At least for Tidal, the URL doesn't match the URI from search, but does seem to work correctly for playback.

**Comment by luka6000** [comment]
any news on this one? Without search there is now way of getting exact URI of the playlist and this seams like a low hanging fruit.

**Reply by OzGav** [reply]
Why aren’t you using search?

**Reply by luka6000** [reply]
> Why aren’t you using search?

There is no search in core
![image](https://github.com/user-attachments/assets/608806dd-e83a-4758-b811-308ed80a6461)

and there is no MAss repo in HACS
![image](https://github.com/user-attachments/assets/73eff549-4518-469e-ba72-4191b28e3e48)

How can I get MAss created playlist URI?

**Reply by OzGav** [reply]
You aren’t using HA 2025.1.x then

**Reply by luka6000** [reply]
Ok, I'm not. But how is search relevant? Sure it allows to search for the playlist uri but why search if one could copy? Same as „provider details”. In May it was a nice small feature to add so I’m just politely asking if there are signs of progress. Being able to point exactly „the playlist” without search sounds like a good idea, don’t you think? And in the era of constantly misunderstood ai voice requests and wrong answers, pinpointing a playlist is this little feature that could make people happy.
And the explanation how to get URI would get much simpler: just copy it.

So are there any signs of this feature going live?
Thank you

**Reply by OzGav** [reply]
No there is not. You can submit a PR to add this.

**Comment by DonGar** [comment]
If you open the playlist through the UI, scroll down to Provider Details
and expand it. That contains a URL that has been working for me.

On Wed, Jan 8, 2025 at 9:33 AM Luka Gra ***@***.***> wrote:

> Why aren’t you using search?
>
> There is no search in core
> image.png (view on web)
> <https://github.com/user-attachments/assets/608806dd-e83a-4758-b811-308ed80a6461>
>
> and there is no MAss repo in HACS
> image.png (view on web)
> <https://github.com/user-attachments/assets/73eff549-4518-469e-ba72-4191b28e3e48>
>
> How can I get MAss created playlist URI?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/music-assistant/discussions/2311#discussioncomment-11775041>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABDGGKNQ46GG2NWCDKEGGD2JVAKFAVCNFSM6AAAAABTDVFCQWVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNZXGUYDIMI>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>


-- 
Don

**Reply by luka6000** [reply]
Are you sure about this? Every link there starts with http and MAss URIs are not http links - as much as I like the idea of identifying http sources as media_id, MAss does not do that.

**Comment by DonGar** [comment]
It works for me, with Tidal.

**Reply by luka6000** [reply]
ok, good for you with Tidal.
For apple music, yt music or internal playlists it doesn't work.
"provider details" is not correct information on MAss URI and MAss is not able to play "provider details" links for any of my providers.

**Comment by LukasGrebe** [comment]
made my way to this board with this issue.
i found myself looking for a `media_id` when calling `action: music_assistant.play_media`

i found the id, as described in the "provider details" at the bottom. however as described this text can not be selected (safari) thus not copied easily.

i when through the develper tools to insepct the state and got a vastly different value `library://track/570` as oppsed to the fileshare details of just a file path.

UX wise, i also initially looked in the 3-dot menu für a "copy track link" or "share"-like option as with most musik provider's native apps.

**Reply by OzGav** [reply]
Unfortunately, you didn’t find the docs. This is way simpler. See the first line here https://www.music-assistant.io/faq/how-to/#get-the-uri

**Reply by fuglphoenix** [reply]
@OzGav, I disagree. It is not way simpler. I don't always know, the name of the album name of an artist is or vice versa.   
So i search through the MASS UI and look at the images of the album covers or other cues to identify what a want. Then i can use search function, that you mention, to find the URI. If i don't have artist and album it is difficult to find the correct URI.   
and it is even more difficult to use the search action, when searching for a playlist. how would you find the URI if the playlist is called: workout or Christmas music.  
however if the URI was available in the 3-dot menu or was selectable in the MASS UI, it would be way simpler.

**Reply by OzGav** [reply]
Well actually you do know because when you navigate to the page you are talking about the album and artist name is in the banner at the top. I didn't mention the search function to you because you don't need it? This request is noted though and will be done when there is time.

**Comment by OzGav** [comment]
In MA 2.4 the provider details section in the Artist, Album and Track views now have a chain link icon which will copy the URI to the clipboard. Note the browser must be accessing the MA frontend via HTTPS for this to work due to browser security rules.

**Reply by rarroyo6** [reply]
I'm sure most HA users are running local copies, and like me are not using HTTPS, so it kind of defeats the purpose of the chain link icon.
Why not just make it so you can select the text and copy it?

**Reply by OzGav** [reply]
This has already been fixed in the latest 2.7 beta

**Comment by petertoth-dev** [comment]
It would be so great to be able to just copy the URL with one click. It's such a tiny development, but it would be a huge plus. It's so hard to go to the original track's/album's/artist's site.  Or at least make the Provider details section open by default (or configurable). 🥹🥹🥹

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
